### PR TITLE
created "private chat 1:Many" for client and slack

### LINKED ISSUE #16: Ability to have one slack channel but private messages
As a client user, I would like to have support chats with anyone in the slack channel, but only see relevant messages for me, so that I can keep my personal chat focused and directed to me.

As a slack chat user, I would like to be able to read through all the client slacks and provide "1 on 1" support, so that I can gain the knowledge and allow the client to feel like they are getting personal attention.

Thoughts?

### PR DESCRIPTION
**Pull request by SxMShaDoW**
- fixed the code to use only spacing
- added another parameter for `clientPrivateChat`
- turned the HTML of slackQuery into a function
- check if the bot user name is in the message before displaying the chat

fixes #16 

Note: I marked where actual "code change" happened, so it isn't as confusing (as its confusing with the spacing change 😢 )

### CODE REVIEW

File: js/slack-chat.js [hunk 1]
```diff
@@ -59,447 +60,463 @@
...
+                                            else
+                                              userName = currentUser.name;
+
+                                            userImg = currentUser.profile.image_48;
+
+                                            break;
+                                          }
+                                        }
+                                      }
+                                    function slackMessageHTML() {
```
Comment (SxMShaDoW) [inline js/slack-chat.js:262]: actual code change here

File: js/slack-chat.js [hunk 2]
```diff
@@ -59,447 +60,463 @@
...
+                                        html += "<div class='msgBox'>"
+                                        html += "<div class='username main'>" + userName + "</div>";
+                                        html += "<div class='message'>" + messageText + "</div>";
+                                        if(typeof moment !== 'undefined')
+                                            html += "<div class='timestamp'>" + moment.unix(resp.messages[i].ts).fromNow() + "</div>";
+                                        html += "</div>";
+                                        html += "</div>";
+                                        return html
+                                    }
+                                    if (window.slackChat._options.clientPrivateChat == true){
```
Comment (SxMShaDoW) [inline js/slack-chat.js:275]: code change here

File: js/slack-chat.js [hunk 3]
```diff
@@ -59,447 +60,463 @@
...
+                                        userImg = message.attachments[0].author_icon;
+                                    }
+
+                                    if(message.fields)
+                                        msgUserId = message.fields[0].value;
+
+                                    var messageText = methods.formatMessage(message.text.trim());
+
+                                    //var messageText = methods.checkforLinks(message);
+                                    function clientMessageHTML () {
```
Comment (SxMShaDoW) [inline js/slack-chat.js:212]: code change here

File: js/slack-chat.js [hunk 4]
```diff
@@ -59,447 +60,463 @@
...
+                                            html += "<div class='username'>" + (msgUserId == options.userId? "You":userName) + "</div>";
+                                        else
+                                            html += "<div class='username'>" + userName + "</div>";
+                                        html += "<div class='message'>" + messageText + "</div>";
+                                        if(typeof moment !== 'undefined')
+                                            html += "<div class='timestamp'>" + moment.unix(resp.messages[i].ts).fromNow() + "</div>";
+                                        html += "</div>";
+                                        html += "</div>";
+                                        return html;
+                                    }
```
Comment (SxMShaDoW) [inline js/slack-chat.js:229]: code change here

File: js/slack-chat.js [hunk 5]
```diff
@@ -32,16 +32,17 @@
         webCache: false,
         privateChannel: false,
         privateChannelId: false,
-				isOpen: false,
-				badgeElement: false,
-				serverApiGateway: "/server/php/server.php",
-				useUserDetails: false,
-				defaultInvitedUsers: []
-	   	};
+        clientPrivateChat: false, //Only show the client their messages and slack messages w/ botUser name in them
```
Comment (SxMShaDoW) [inline js/slack-chat.js:35]: code change here

**Comment by rotatordisk92** [comment]
@SxMShaDoW  Good idea.
I tested out your PR, worked for me.

**Comment by arjunmat** [comment]
@SxMShaDoW - can you please rebase this with master, so I can merge it in. Had to push in a fix for links in v1.5.1

**Comment by SxMShaDoW** [comment]
@improvi - Rebased and re-tested to make sure nothing got missed with my checkin.

**Comment by SxMShaDoW** [comment]
@improvi - should be good to go. let me know if you have anymore questions.

**Comment by SxMShaDoW** [comment]
@improvi any update on this? its been 11 days or so.

**Comment by arjunmat** [comment]
can't get this working. when the clientPrivateChat is set to true, no messages are being displayed in the chat window. Anything else i need to do?

**Comment by SxMShaDoW** [comment]
It's looking for you to specify the user of the bot for the messages to
show.

**Comment by SxMShaDoW** [comment]
Create a user lets say "test_user". send a message via the chat. The "test_user" should be able to see the message they just sent. The slack channel that monitors it will see the "test_user" send a message. 
- Negative test: From slack send a message back (it should _not_ show on the "test_user" side). 
- Positive test: From slack send a message back with "test_user" in the message of the response. This should show on the chat now and allows the user that connected to it to see the message.

**Comment by SxMShaDoW** [comment]
just going to close this as it doesn't look like you will be adding this functionality. will just use my fork for the project.

**Pull request closed without merging**